Fastboot device detection fix, Linux terminal option settings, logging improvements

zraexy opened the pull request:
These commits resolve a couple of issues I ran into using PixelFlasher.

The fastboot issue was triggered by plugging in my Pixel on the same USB bus that my laptop fingerprint reader is on.

The shell/terminal changes were made to support using KDE Plasma and NixOS.
KDE konsole uses "-e" and "--workdir" for its options ("-e" has different argument splitting behavior in gnome-terminal).
NixOS doesn't have a global "/bin/bash".

Ideally there would be an error message if "Linux Shell" is not set, but I opted to instead just supply defaults because it will at least log an error. The current behavior for running commands doesn't even provide any notification that there is an issue. Additionally, even if "Linux Shell" is set, it doesn't mean that the default option arguments are correct for the specified shell, and I assume you don't want to force people to manually provide the options when using gnome-terminal.

Thank you for building this!

badabing2005 commented:
> The shell/terminal changes were made to support using KDE Plasma and NixOS.
KDE konsole uses "-e" and "--workdir" for its options ("-e" has different argument splitting behavior in gnome-terminal).
NixOS doesn't have a global "/bin/bash".
>
> Ideally there would be an error message if "Linux Shell" is not set, but I opted to instead just supply defaults because it will at least log an error. The current behavior for running commands doesn't even provide any notification that there is an issue. Additionally, even if "Linux Shell" is set, it doesn't mean that the default option arguments are correct for the specified shell, and I assume you don't want to force people to manually provide the options when using gnome-terminal.

The idea of Linux shell / File manager settings is simple, people know their own shell and parameters required to launch what they like how they like.

They set the entire command (with arguments) into the settings.
Simple as that, rather than split the command, arguments, directory ... into separate fields.
There is `My Tools` section for that, have you checked?
I don't want to overcrowd the settings window, which is already crowded enough.

I like your changes, specially the part about making it more platform agnostic, but having extra UI widgets for arguments / directory in settings window, I'd rather do without.

Thanks

zraexy commented:
I have removed the separate settings.

Currently, anyone who has explicitly specified a "Linux Shell" will have their setup break because the command option argument must now be manually included. Perhaps the correct thing to do would be to change the attribute name in the JSON config so that it becomes unset?

I have also removed the changes that make gnome-terminal the default in all cases, although, it does seem strange to default to using gnome-terminal when opening a shell in a given directory, but not to do so for launching ADB Shell & Scrcpy.
